proxy/envoy/discovery.go

```diff
@@ -523,22 +509,48 @@ func (ds *DiscoveryService) ListRoutes(request *restful.Request, response *restf
 	writeResponse(response, out)
 }
 
+// ListListeners responds to LDS requests
+func (ds *DiscoveryService) ListListeners(request *restful.Request, response *restful.Response) {
+	key := request.Request.URL.String()
+	out, cached := ds.ldsCache.cachedDiscoveryResponse(key)
+	if !cached {
```

Comment:
Consider inverting the logic to reduce indentation:
if cached { 
     write 
     return 
}

// the long logic here